reana_db/utils.py

```diff
@@ -258,11 +258,79 @@ def update_users_disk_quota(user=None, bytes_to_sum: Optional[int] = None) -> No
             Session.commit()
 
 
+def update_workflow_cpu_quota(workflow) -> None:
```

Comment:
An example where I cannot use type hint due to circular import issue.

```suggestion
def update_workflow_cpu_quota(workflow: Workflow) -> None:
```

Related to https://github.com/reanahub/reana/issues/557#issuecomment-978070904

Reply:
*thought:* Circular imports indicate that the code in `modes.py` and `utils.py` is too coupled and requires reorganization. Splitting code into: 

- `utils.py` file with low-level functions that do not depend on anything from reana_db.models (e.g, `build_workspace_path`)

- `quota.py` file with functions that are built on top of DB models and low-level functions (e.g, `update_workflow_cpu_quota`) 

- move `workflow_status_change_listener` function from `models.py` to `event_listener.py `

would solve a problem. 

There will be a clear dependency path: `utils.py` => `models.py` => `quotas.py` => `event_listeners.py`

This requires quite some work so definitely not for this PR. If the idea sounds interesting we can create an issue for this.